src/axiosInstance/index.ts

```diff
@@ -1,6 +1,8 @@
 import axios, { AxiosError } from 'axios'
 import { BACKEND_BASE_URL } from '../utils/settings'
 
+const baseURL = process.env.NODE_ENV === 'production' ? BACKEND_BASE_URL : 'http://localhost:3000'
```

Comment:
I think it is unnecessary to check this conditionally. You could just do
```
const baseURL = BACKEND_BASE_URL;
```
And then if you have the correct env variable it will work. This is just an observation. If you have reason to do it this way, plz go ahead.